Source/EasyNetQ/Logging/ILogger.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace EasyNetQ.Logging
+{
+    public interface ILogger
+    {
+        bool Log(LogLevel logLevel, Func<string> messageFunc, Exception exception = null, params object[] formatParameters);
```

Comment:
In principle, we can really get by with one method.

Reply:
yes, and you will see that internally there are a whole set of extension written on top of that single method, but I basically use that interface to avoid changing to much of the actual internals of easynetq, it's a trade off to have both world with minimum code change.